samples/ask-akka-agent/src/main/java/akka/ask/agent/application/StructuredAskAkkaAgent.java

```diff
@@ -0,0 +1,35 @@
+package akka.ask.agent.application;
+
+import akka.javasdk.agent.Agent;
+import akka.javasdk.annotations.AgentDescription;
+import akka.javasdk.annotations.ComponentId;
+
+@ComponentId("structured-ask-akka-agent")
+@AgentDescription(name = "Akka Design Specialist", description = "Expert in Akka design questions")
+public class StructuredAskAkkaAgent extends Agent {
+  public record Response(String summary, String akkaComponent) {}
+
+  private final String sysMessage = """
+      You are a specialized agent that helps with Akka design question.
+      
+      IMPORTANT: You must respond with a valid JSON object that follows this structure:
+        {
+          "summary": "Brief overview of the trip",
+          "akkaComponent": "The name of the Akka component to use"
+        }
+      
+      Do not include any explanations or text outside of the JSON structure.
+      If you are unsure, say the following in the summary field:
+      Sorry, I don't know how to help with that.
+      """; // <1>
+
+  public Effect<Response> ask(String question) {
+    return effects()
+        .modelProvider(ModelProvider.OPEN_AI)
+        .modelApiKey(System.getenv("OPENAI_API_KEY"))
+        .modelName("gpt-4o-mini")
+        .systemMessage(sysMessage)
+        .userMessage(question)
+        .thenReplyAs(Response.class);
```

Comment:
For structured responses, I wonder if we should have some kind of result type. In case invalid json is returned, and you can get an error with the raw response.